sentry/src/main/java/io/sentry/SentryTracer.java

```diff
@@ -97,7 +124,19 @@ private ISpan createChild(
     Objects.requireNonNull(parentSpanId, "parentSpanId is required");
     Objects.requireNonNull(operation, "operation is required");
     final Span span =
-        new Span(root.getTraceId(), parentSpanId, this, operation, this.hub, timestamp);
+        new Span(
+            root.getTraceId(),
+            parentSpanId,
+            this,
+            operation,
+            this.hub,
+            timestamp,
+            __ -> {
+              final FinishStatus finishStatus = this.finishStatus;
+              if (finishStatus.isFinishing) {
+                finish(finishStatus.spanStatus);
+              }
```

Comment:
L: I would just suggest extracting this to a function, let's say in the future we have another condition to finish, or other places this verification should be done.
It would help with maintenance.

Maybe SentryTracer itself could implement SpanListener.

Reply:
not every span has the `waitForChildren` feat, so `Span` getting the listener as ctor makes sense to me, but its an implementation detail anyway